prediction_market_agent_tooling/tools/perplexity/perplexity_client.py

```diff
@@ -0,0 +1,75 @@
+import os
+from typing import Any, Dict, List, Optional
+
+import httpx
+
+from .perplexity_models import (
+    PerplexityModelSettings,
+    PerplexityRequestParameters,
+    PerplexityResponse,
+)
+
+
+class PerplexityModel:
+    def __init__(
+        self,
+        model_name: str,
+        *,
+        api_key: Optional[str] = None,
+        completition_endpoint: str = "https://api.perplexity.ai/chat/completions",
+    ) -> None:
+        self.model_name: str = model_name
+        self.api_key: str | None = api_key or os.environ.get("PERPLEXITY_API_KEY")
```

Comment:
```suggestion
        self.api_key: SecretStr = api_key or APIKeys().perplexity_api_key
```

APIKeys automatically loads `.env` file, so we don't need to think about that, and it will also automatically raise meaningful error, if the key is not available in the env

Reply:
Fixed